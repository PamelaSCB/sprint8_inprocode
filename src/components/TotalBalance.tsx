const TotalBalance = () => {
    return (
      <div>
        <h1>Total Balance</h1>
      </div>
    );
  };
  
  export default TotalBalance;
  