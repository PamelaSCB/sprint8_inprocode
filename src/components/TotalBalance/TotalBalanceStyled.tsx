import { styled } from 'styled-components';

export const TotalBalanceStyled = styled.div`
width: 40%;
padding: 1.5rem;
border-radius:1rem;
background-color: #ec765c ;
color: #fff;
display : flex;
justify-content: space-between;

.totalBalance {
    display: flex;
    flex-direction: column;

    span {
        font-size: 1.2rem;
    }
    
    p {
        margin :1rem 0 0 0;
        strong{
            font-size: 2rem;
        }
    }
}

.navigation-arrows {
    display: flex;
    flex-direction: row;

    img {
        width: 2.5rem;
        height: 2.5rem;
        cursor: pointer;
     }
}


`
