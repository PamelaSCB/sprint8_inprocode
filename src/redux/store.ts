import {AnyAction,configureStore} from "@reduxjs/toolkit";
import { ToolkitStore } from "@reduxjs/toolkit/dist/configureStore";

import expensesTodayReducer, {ExpensesTodayState} from "./slices/expensesTodaySlices";
import graphExpensesReducer, {GraphExpensesState} from "./slices/graphExpensesSlices.ts";
import totalBalanceReducer,{TotalBalanceState} from "./slices/totalBalanceSlices";
import variationYesterdayVsTodayReducer, {VariationYesterdayVsTodayState} from "./slices/variationYesterdayVsTodaySlices";
import currencyReducer, {CurrencyState} from "./slices/currencySlices";
import weekNumberReducer,{WeeknumberState} from "./slices/weekNumberSlices";

export interface Store {
  totalBalance: TotalBalanceState; 
  currency: CurrencyState;
  weekNumber: WeeknumberState;
  expensesToday: ExpensesTodayState;
  graphExpenses: GraphExpensesState;
  variationYesterdayVsToday: VariationYesterdayVsTodayState;
}

export const store:ToolkitStore<Store, AnyAction> = configureStore({
  reducer: {
    expensesToday: expensesTodayReducer,
    graphExpenses: graphExpensesReducer,
    totalBalance: totalBalanceReducer,
    variationYesterdayVsToday: variationYesterdayVsTodayReducer,
    currency : currencyReducer,
    weekNumber : weekNumberReducer,
  },
});
export default store; 

