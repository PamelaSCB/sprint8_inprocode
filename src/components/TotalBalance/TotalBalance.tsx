import { useDispatch, useSelector } from "react-redux";
import { Store } from "../../redux/store";

import { useEffect } from "react";
import { useTranslation } from "react-i18next";

import { changeTotalBalance } from "../../redux/slices/totalBalanceSlices";
import { TotalBalanceStyled } from "./TotalBalanceStyled";

function TotalBalance(): JSX.Element {
  const dispatch = useDispatch();
  const { t } = useTranslation();

  const selectGraphExpenses = (state: Store) => state.graphExpenses;
  const selectTotalBalance = (state: Store) => state.totalBalance;
  const selectCurrency = (state: Store) => state.currency;

  const graphExpenses = useSelector(selectGraphExpenses);
  const totalBalance = useSelector(selectTotalBalance);
  const currency = useSelector(selectCurrency);

  useEffect(() => {
    const total: number =
      graphExpenses.monday +
      graphExpenses.tuesday +
      graphExpenses.wednesday +
      graphExpenses.thursday +
      graphExpenses.friday +
      graphExpenses.saturday +
      graphExpenses.sunday;

    dispatch(changeTotalBalance(total));
  }, [graphExpenses, changeTotalBalance]);

  return (
    <TotalBalanceStyled>
      <div className="totalBalance">
        <span>
          <small>{t("main.totalBalance")}</small>
        </span>
        <p>
          <strong>
            {totalBalance.totalBalance} {currency.currency}
          </strong>
        </p>
      </div>
    </TotalBalanceStyled>
  );
}

export default TotalBalance;
