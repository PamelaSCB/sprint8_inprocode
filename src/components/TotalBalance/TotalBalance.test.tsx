import {describe,test,expect} from "vitest";
import TotalBalance from "./TotalBalance";
import {store} from "../../redux/store";
import {Provider} from "react-redux";


describe("TotalBalance", () => {
    test("should render correctly", () => {
        const tree = (
            <Provider store={store}>
                <TotalBalance />
            </Provider>
        );
        expect(tree).toMatchSnapshot();

        
         });
});