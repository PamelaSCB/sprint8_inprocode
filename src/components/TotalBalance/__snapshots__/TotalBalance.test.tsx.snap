// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TotalBalance > should render correctly 1`] = `
<Provider
  store={
    {
      "@@observable": [Function],
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <TotalBalance />
</Provider>
`;
